implementations/micrometer-registry-stackdriver/src/main/java/io/micrometer/stackdriver/StackdriverMeterRegistry.java

```diff
@@ -328,6 +350,13 @@ private TimeSeries createTimeSeries(Meter.Id id, TypedValue typedValue, MetricDe
             String metricType = metricType(id, statistic);
 
             Map<String, String> metricLabels = getConventionTags(id).stream()
+                    // label value cannot be more than 1024 characters. See https://cloud.google.com/monitoring/quotas#custom_metrics_quotas
+                    .peek(tag -> {
+                        if (tag.getValue().length() >= 1024) {
+                            logger.warn("Filtered out tag {}, for metric {}, because its value exceeded the max length of 1024 characters", tag.getKey(), metricType);
```

Comment:
Normally we truncate to satisfy length requirements like this. This is done in Micrometer in the `NamingConvention` for the MeterRegistry. For example, you can see we already truncate meter names that are too long: https://github.com/micrometer-metrics/micrometer/blob/a386d1b23173d61b563288faf3b4e5195fe0a965/implementations/micrometer-registry-stackdriver/src/main/java/io/micrometer/stackdriver/StackdriverNamingConvention.java#L52-L59

If we override the `tagValue` method, we can do the same for the tag value there.